x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/routes.ts

```diff
@@ -293,6 +296,40 @@ const dataStreamSettingsRoute = createDatasetQualityServerRoute({
   },
 });
 
+const checkAndLoadIntegrationRoute = createDatasetQualityServerRoute({
+  endpoint: 'GET /internal/dataset_quality/data_streams/{dataStream}/integration/check',
+  params: t.type({
+    path: t.type({
+      dataStream: t.string,
+    }),
+  }),
+  options: {
+    tags: [],
+  },
+  async handler(resources): Promise<CheckAndLoadIntegrationResponse> {
+    const { context, params, plugins, logger } = resources;
+    const { dataStream } = params.path;
+    const coreContext = await context.core;
+
+    // Query dataStreams as the current user as the Kibana internal user may not have all the required permissions
+    const esClient = coreContext.elasticsearch.client.asCurrentUser;
+
+    const fleetPluginStart = await plugins.fleet.start();
+    const packageClient = fleetPluginStart.packageService.asInternalUser;
+
+    const integration = await checkAndLoadIntegration({
+      esClient,
+      packageClient,
+      logger,
+      dataStream,
+    });
+
+    logger.info('aj-integration' + JSON.stringify(integration));
```

Comment:
Not sure, but this seems a debugging leftover

Reply:
Indeed, good 👍